quic-client/src/lib.rs

```diff
@@ -143,7 +156,23 @@ impl QuicConfig {
         ipaddr: IpAddr,
     ) -> Result<(), RcgenError> {
         let (cert, priv_key) = new_self_signed_tls_certificate(keypair, ipaddr)?;
-        self.client_certificate = Arc::new(QuicClientCertificate {
+        self.addr = ipaddr;
+
+        let mut cert_guard = self.client_certificate.write().unwrap();
+
+        *cert_guard = Arc::new(QuicClientCertificate {
+            certificate: cert,
+            key: priv_key,
+        });
+        Ok(())
+    }
+
+    pub fn update_keypair(&self, keypair: &Keypair) -> Result<(), RcgenError> {
```

Comment:
This update works only when a new connection pool is created. How do we have handle existing connection cached?

Reply:
See connection-cache/src/connection_cache.rs line 141

Reply:
I understand that, but I think it won't fix the existing connection's endpoints. We would similar things you did to the server side: like 

        let (config, _) = configure_server(key, self.gossip_host)?;
        self.endpoint.set_server_config(Some(config));
        Ok(())